### PR TITLE
feat: add option to generate random content for new resources

### PR DESCRIPTION
**Pull request by olensmar**
added the following checkbox to the New Resource dialog (checking it disables the template selection):

image

Random content is generated - although I'm not sure it's always so helpful...

image

## Checklist

- [X] tested locally
- [X] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: src/components/organisms/NewResourceWizard/NewResourceWizard.tsx [hunk 1]
```diff
@@ -567,12 +598,18 @@ const NewResourceWizard = () => {
...
+          </Checkbox>
+        </Form.Item>
+
         <Form.Item
           name="selectedResourceId"
           label="Select existing resource as template"
           initialValue={SELECT_OPTION_NONE}
         >
-          <Select showSearch>
+          <Select showSearch disabled={generateRandom === true}>
```
Comment (topliceanurazvan) [inline src/components/organisms/NewResourceWizard/NewResourceWizard.tsx:612]:
```suggestion
          <Select showSearch disabled={generateRandom}>
```

File: src/redux/services/unsavedResource.ts [hunk 2]
```diff
@@ -35,16 +35,30 @@ export function createUnsavedResource(
...
-      ...jsonTemplate,
-      apiVersion: input.apiVersion,
-      kind: input.kind,
-      metadata: {
-        ...(jsonTemplate.metadata || {}),
-        name: input.name,
-        namespace: input.namespace,
-      },
-    };
+    if (jsonTemplate.kind && jsonTemplate.apiVersion) {
```
Comment (topliceanurazvan) [inline src/redux/services/unsavedResource.ts:38]: We could use directly this here so we don't duplicate the object:

```
newResourceContent = {
        ...jsonTemplate,
        apiVersion: input.apiVersion,
        kind: input.kind,
        metadata: {
          ...(jsonTemplate.metadata || {}),
          name: input.name,
          namespace: input.namespace,
        },
        ...(jsonTemplate.kind && jsonTemplate.apiVersion && {...jsonTemplate}),
      };
```

**Review by topliceanurazvan — COMMENTED** [review]
Looks good 👍🏻 left a couple of comments...

File: src/redux/services/unsavedResource.ts (hunk 2 above)
Comment (olensmar) [reply src/redux/services/unsavedResource.ts:38]: the problem I had with putting jsonTemplate first was that the apiVersion/kind/metadata properties were added to the end of the document - which was weird in the editor

**Review by topliceanurazvan — APPROVED** [review]

**Pull request merged**